bypros/api.go

```diff
@@ -0,0 +1,213 @@
+package bypros
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gorilla/websocket"
+	"go.dedis.ch/cothority/v3"
+	"go.dedis.ch/cothority/v3/skipchain"
+	"go.dedis.ch/onet/v3"
+	"go.dedis.ch/onet/v3/network"
+	"go.dedis.ch/onet/v4/log"
+	"go.dedis.ch/protobuf"
+	"golang.org/x/xerrors"
+)
+
+var overlayClient OverlayClient = onetOverlay{
+	onet.NewClient(cothority.Suite, ServiceName),
+}
+
+// OverlayClient defines the primitives needed by our client. That's all we need
+// from the outer world.
+type OverlayClient interface {
+	SendProtobuf(dst *network.ServerIdentity, msg interface{}, ret interface{}) error
+	OpenWS(url string) (WsHandler, error)
+}
+
+// WsHandler defines the primitives to handle a websocket connection.
+type WsHandler interface {
+	Close() error
+	Write(messageType int, data []byte) error
+	Read() (messageType int, p []byte, err error)
+}
+
+// NewClient creates a new proxy client
+func NewClient() *Client {
+	return &Client{OverlayClient: overlayClient}
+}
+
+// Client defines a proxy client
+//
+// - implements OverlayClient
+type Client struct {
+	OverlayClient
+}
+
+// Follow sends a request to start following a node, ie. listening to new blocks
+// and updating the database accordingly.
+func (c *Client) Follow(host, target *network.ServerIdentity,
```

Comment:
Shouldn't this take a full roster?

Reply:
no. The service allows you to follow a specific node.

Reply:
That doesn't really make sense. A ByzCoin chain is not defined by the nodes, but rather by the genesis-ID and all the forward-links. Nodes can come and go, they might be down. So if you have this `Follow` with only one node, and then only follow that node, the proxy might be quite unstable.